### ISSUE TITLE
NullPointerException: Attempt to get length of null array (InstallReferrer.java:252)

### ISSUE THREAD

**Issue by RenanLukas** [adjust/android_sdk#320]
Hi

We detected a crash in production (version 4.12.3 of the SDK). Do you know what could be the cause?

```
Fatal Exception: java.lang.NullPointerException: Attempt to get length of null array
       at com.adjust.sdk.InstallReferrer.invoke(InstallReferrer.java:252)
       at java.lang.reflect.Proxy.invoke(Proxy.java:393)
       at $Proxy7.onInstallReferrerServiceDisconnected(Unknown Source)
       at com.android.installreferrer.api.InstallReferrerClientImpl$InstallReferrerServiceConnection.onServiceDisconnected(InstallReferrerClientImpl.java:230)
       at android.app.LoadedApk$ServiceDispatcher.doDeath(LoadedApk.java:1340)
       at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:1354)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:158)
       at android.app.ActivityThread.main(ActivityThread.java:7224)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by uerceg** [comment]
Hi @RenanLukas 

Thanks for reporting. Can you maybe share some additional information on how often did you notice this crash in production and on which devices (if possible their API levels)?

**Comment by RenanLukas** [comment]
Hey @uerceg 
Thanks for the quick answer.
We released our first production version with Adjust yesterday.
So far we only noticed this crash on a Samsung Galaxy S5 device running on API 23.

**Comment by uerceg** [comment]
@RenanLukas 

Thanks for this information, we're investigating this in attempt to reproduce it and see what might be causing this. Will keep you posted in here as soon as we have some updates on the issue.

**Comment by uerceg** [comment]
Also, in case you notice this crash incoming maybe apart from this single device, would be great if you can share information about additional devices in here.

**Comment by uerceg** [comment]
Hi @RenanLukas 

Indeed tricky issue with no clear way of being reproduced, regardless of our attempts. Never the less, actions on our side could have been taken to gracefully handle it's occurrence and we have released today v4.12.4 which should hopefully address this issue: https://github.com/adjust/android_sdk/releases/tag/v4.12.4

I will close this issue, but feel free to reopen it if you encounter this issue again with SDK v4.12.4++.

Thanks one more time for the report.

Cheers!
